samples/go/models/jsonrpc.go

```diff
@@ -0,0 +1,46 @@
+package models
+
+// JSONRPCMessageIdentifier represents the base interface for identifying JSON-RPC messages
+type JSONRPCMessageIdentifier struct {
+	// ID is the request identifier. Can be a string, number, or null.
+	// Responses must have the same ID as the request they relate to.
+	// Notifications (requests without an expected response) should omit the ID or use null.
+	ID interface{} `json:"id,omitempty"`
+}
+
+// JSONRPCMessage represents the base interface for all JSON-RPC messages
+type JSONRPCMessage struct {
+	JSONRPCMessageIdentifier
+	// JSONRPC specifies the JSON-RPC version. Must be "2.0"
+	JSONRPC string `json:"jsonrpc,omitempty"`
+}
+
+// JSONRPCRequest represents a JSON-RPC request object base structure
+type JSONRPCRequest struct {
+	JSONRPCMessage
+	// Method is the name of the method to be invoked
+	Method string `json:"method"`
+	// Params are the parameters for the method
+	Params interface{} `json:"params,omitempty"`
```

Comment:
Excellent catch here, the JSONRPC spec allows `params` to be both a `slice` and a `map`.